lib/std/math/cos.zig

```diff
@@ -15,109 +20,135 @@ const expect = std.testing.expect;
+            }
+        }
+    }
+    if (ix <= 0x40e231d5) { // |x| ~<= 9*pi/4
+        if (ix > 0x40afeddf) { // |x| ~> 7*pi/4
+            return kernel.__cosdf(if (sign) x + c4pio2 else x - c4pio2);
+        } else {
+            if (sign) {
+                return kernel.__sindf(-x - c3pio2);
+            } else {
+                return kernel.__sindf(x - c3pio2);
+            }
+        }
     }
 
-    var sign = false;
-    x = math.fabs(x);
+    // cos(Inf or NaN) is NaN
+    if (ix >= 0x7f800000) {
+        return x - x;
+    }
 
-    var y = math.floor(x * m4pi);
-    var j = @floatToInt(I, y);
+    var y: f64 = undefined;
+    const n = __rem_pio2f(x, &y);
+    return switch (n & 3) {
+        0 => kernel.__cosdf(y),
+        1 => kernel.__sindf(-y),
+        2 => -kernel.__cosdf(y),
+        else => kernel.__sindf(y),
+    };
+}
 
-    if (j & 1 == 1) {
-        j += 1;
-        y += 1;
+pub fn cos64(x: f64) f64 {
+    var ix = @bitCast(u64, x) >> 32;
+    ix &= 0x7fffffff;
+
+    // |x| ~< pi/4
+    if (ix <= 0x3fe921fb) {
+        if (ix < 0x3e46a09e) { // |x| < 2**-27 * sqrt(2)
+            // raise inexact if x!=0
+            math.doNotOptimizeAway(x + 0x1p120);
+            return 1.0;
+        }
+        return kernel.__cos(x, 0);
     }
 
-    j &= 7;
-    if (j > 3) {
-        j -= 4;
-        sign = !sign;
+    // cos(Inf or NaN) is NaN
+    if (ix >= 0x7ff00000) {
+        return x - x;
     }
-    if (j > 1) {
-        sign = !sign;
-    }
-
-    const z = ((x - y * pi4a) - y * pi4b) - y * pi4c;
-    const w = z * z;
-
-    const r = if (j == 1 or j == 2)
-        z + z * w * (S5 + w * (S4 + w * (S3 + w * (S2 + w * (S1 + w * S0)))))
-    else
-        1.0 - 0.5 * w + w * w * (C5 + w * (C4 + w * (C3 + w * (C2 + w * (C1 + w * C0)))));
 
-    return if (sign) -r else r;
+    var y: [2]f64 = undefined;
+    const n = __rem_pio2(x, &y);
+    return switch (n & 3) {
+        0 => kernel.__cos(y[0], y[1]),
+        1 => -kernel.__sin(y[0], y[1], 1),
+        2 => -kernel.__cos(y[0], y[1]),
+        else => kernel.__sin(y[0], y[1], 1),
+    };
 }
 
 test "math.cos" {
-    try expect(cos(@as(f32, 0.0)) == cos_(f32, 0.0));
-    try expect(cos(@as(f64, 0.0)) == cos_(f64, 0.0));
+    try expect(cos(@as(f32, 0.0)) == cos32(0.0));
+    try expect(cos(@as(f64, 0.0)) == cos64(0.0));
 }
 
 test "math.cos32" {
-    const epsilon = 0.000001;
-
-    try expect(math.approxEqAbs(f32, cos_(f32, 0.0), 1.0, epsilon));
-    try expect(math.approxEqAbs(f32, cos_(f32, 0.2), 0.980067, epsilon));
-    try expect(math.approxEqAbs(f32, cos_(f32, 0.8923), 0.627623, epsilon));
-    try expect(math.approxEqAbs(f32, cos_(f32, 1.5), 0.070737, epsilon));
-    try expect(math.approxEqAbs(f32, cos_(f32, -1.5), 0.070737, epsilon));
-    try expect(math.approxEqAbs(f32, cos_(f32, 37.45), 0.969132, epsilon));
-    try expect(math.approxEqAbs(f32, cos_(f32, 89.123), 0.400798, epsilon));
+    const epsilon = 0.00001;
```

Comment:
I needed to reduce these epsilons since the calculation method changed and the test cases otherwise would need adjustment.